Dockerfile

```diff
@@ -14,40 +14,47 @@
 
 ##
 # Build stage
+# Use github action runner cached images to avoid being rate limited
+# https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images
 ## 
-FROM rust:1-buster AS build-env
+FROM buildpack-deps:buster AS build-env
 
 # Install dependencies
-RUN apt-get update && apt-get install --no-install-recommends -y build-essential clang ocl-icd-opencl-dev cmake
+RUN apt-get update && apt-get install --no-install-recommends -y build-essential clang ocl-icd-opencl-dev cmake ca-certificates curl
+RUN update-ca-certificates
+
+# Install rustup
+# https://rustup.rs/
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+ENV RUSTFLAGS="-Ctarget-feature=+avx2,+fma"
 
 WORKDIR /usr/src/forest
 COPY . .
 
-# Grab the correct toolchain
-RUN rustup toolchain install nightly
-
-ENV RUSTFLAGS="-Ctarget-feature=+avx2,+fma"
-
 # Install Forest
 RUN make install
 
-# strip symbols to make executable smaller
-RUN strip /usr/local/cargo/bin/forest /usr/local/cargo/bin/forest-cli
-
 ##
 # Prod image for forest binary
+# Use github action runner cached images to avoid being rate limited
+# https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images
 ##
-FROM debian:10-slim
+FROM ubuntu:20.04
 
 # Link package to the repository
 LABEL org.opencontainers.image.source https://github.com/chainsafe/forest
 
+ENV DEBIAN_FRONTEND="noninteractive"
 # Install binary dependencies
 RUN apt-get update && apt-get install --no-install-recommends -y ocl-icd-opencl-dev libssl1.1 ca-certificates libcurl4
```

Comment:
libcurl4 should no longer be needed after https://github.com/ChainSafe/forest/pull/2102 was merged

Reply:
Nvm it's still needed. But the new verification step is capable of finding that now! 
https://github.com/ChainSafe/forest/actions/runs/3336256732/jobs/5521188588#step:9:6

Reply:
Beautiful!